index.js

```diff
@@ -4,21 +4,42 @@ var sublevel = require('subleveldown')
 var through = require('through2')
 
 /**
-* Initialize the `townshipToken` module
+* Initialize the `townshipToken` module. You can choose to specify either `secret` or `publicKey` and `privateKey` for signatures. To generate a keypair you can use [these commands](https://gist.github.com/maxogden/62b7119909a93204c747633308a4d769).
 * @name townshipToken
 * @param {object} db – An instance of a leveldb via [level](https://https://github.com/Level/level)
 * @param {object} options
-* @param {string} options.secret – Secret used for signing tokens
+* @param {string} options.algorithm – **Optional.** JWA algorithm, default is `HS256` (HMAC/SHA256 with secret). You must specify your key type if using a keypair.
+* @param {string} options.secret – **Optional.** Secret used for signing and verifying tokens
+* @param {string} options.publicKey – **Optional.** Public key used to sign tokens
+* @param {string} options.privateKey – **Optional.** Private key used to verify tokens
+
 * @example
+* // using a secret
 * var level = require('level')
 * var townshipToken = require('township-token')
 *
 * var db = level('./db')
 * var tokens = townshipToken(db, { secret: process.env.TOWNSHIP_SECRET })
+*
+* @example
+* // using a keypair
+* var tokens = townshipToken(db, {
+*   algorithm: 'ES512',
+*   public: `-----BEGIN PUBLIC KEY-----
+* blahblah
+* -----END PUBLIC KEY-----`,
+*   private: `-----BEGIN EC PRIVATE KEY-----
+* blahblah
+* -----END EC PRIVATE KEY-----`
+* })
 **/
+
 module.exports = function townshipToken (maindb, options) {
   options = options || {}
-  var secret = options.secret || 'not a secret'
+  var secret = options.secret
+  var publicKey = options.publicKey
+  var privateKey = options.privateKey
+  var algorithm = options.algorithm
```

Comment:
annoying that `jsonwebtoken` doesnt allow you to pass in an options object, but rather makes you do the `secretOrPublicOrPrivateKey` magic string thing. so we have to do lots of options checking to wrap their api